### ISSUE TITLE
Preserve mid-edit changes when sharing to simpletask

### ISSUE THREAD

**Issue by smichel17** [mpcjanssen/simpletask-android#717]
Repro steps:

1. Make sure Settings > Features > Edit shared tasks is enabled
2. Edit a task. We'll call it task 1.
3. Make some changes. We'll call this task 1a.
4. Press the home button without saving or discarding changes
5. Share a new task to simpletask. We'll call it task 2

Expected result: Show add/edit task screen with tasks 1a and 2.
Actual result: Show add/edit task screen with tasks 1 and 2.

Fixing #716 will make this easier to fix; we should do that first.

**Comment by smichel17** [comment]
The solution to this is to set `FLAG_ACTIVITY_SINGLE_TOP` as a flag on the intent used to launch the addtask activity [(source)](https://developer.android.com/reference/android/content/Intent.html#FLAG_ACTIVITY_SINGLE_TOP). I'm not 100% sure how this works, but I think it causes the intent to be delivered to the `onNewIntent` function if the activity is already running, so we'll need to override that with the code for adding the shared task to the existing screen.

**Comment by mpcjanssen** [comment]
Properly fixing this is fairly complex. It seems to require updating the underlying tasks at every edit. And in some way merging the background shared tasks. I am beginning to think it might be easier to not allow editing of background tasks but instead use to tag functionality to allow easier changes.

**Comment by smichel17** [comment]
Is there a way to just grab the prior contents of the textview and append to that, since after #716 we'd be returning to a pre-existing activity instead of re-creating a new one?

**Comment by mpcjanssen** [comment]
I don't think so. The actual tasks need to be available to track the updates so just the text is not enough.

**Comment by smichel17** [comment]
Second option: create a `editedTodoItem` which contains a todoitem and the latest revision. Save it in `onPause()`, not each time you exit.

**Comment by mpcjanssen** [comment]
I'll do some experimenting in the standalone branch.

**Comment by mpcjanssen** [comment]
See #721.

**Comment by smichel17** [comment]
Closing in favor of #721
